pkg/healthcheck/healthcheck.go

```diff
@@ -109,6 +115,35 @@ func (hc *simpleHealthChecker) waitForConsole(ctx context.Context, cs *acsapi.Op
 	}
 }
 
+func newCoreClient(ctx context.Context, config *v1.Config) (*corev1client.CoreV1Client, error) {
```

Comment:
combine newCoreClient and newAppClient.  Use `kc, err := kubernetes.NewForConfig(restconfig)`, and then use kc.AppsV1() where you were using your appclient and kc.CoreV1() where you use your coreclient

Reply:
Thanks, will look at this.